service/history/workflow/update/registry.go

```diff
@@ -358,6 +380,10 @@ func (r *registry) Send(
 func (r *registry) Clear() {
 	r.Abort(AbortReasonRegistryCleared)
 
+	for _, f := range r.onClear {
+		f()
+	}
```

Comment:
I am not sure about it. `cancel()` cancels subscription created by `Subscribe()` which is called once per process? `r.Clear()` is called many times. Wouldn't it stop getting notification after first `Clear()` call?

Reply:
Maybe it should be called `Destroy` instead of `Clear`. Clear implies it can be used after.